js_modules/dagit/packages/core/src/assets/AssetEventMetadataEntriesTable.tsx

```diff
@@ -0,0 +1,92 @@
+import {Box, Caption, Colors, Icon, Mono} from '@dagster-io/ui';
+import moment from 'moment';
+import React from 'react';
+import {Link} from 'react-router-dom';
+import styled from 'styled-components/macro';
+
+import {MetadataEntry} from '../metadata/MetadataEntry';
+import {titleForRun} from '../runs/RunUtils';
+
+import {AssetMaterializationFragment} from './types/AssetMaterializationFragment';
+import {AssetObservationFragment} from './types/AssetObservationFragment';
+
+/**
+ * This component shows the metadata entries attached to an Asset Materialization or Observation event.
+ * AssetNodes also have definition-time metadata, which is unrelated to this event metadata.
+ */
+export const AssetEventMetadataEntriesTable: React.FC<{
+  event: AssetObservationFragment | AssetMaterializationFragment | null;
+  observations?: (AssetObservationFragment | AssetMaterializationFragment)[];
+}> = ({event, observations}) => {
+  if (!event || (!event.metadataEntries.length && !observations?.length)) {
+    return <Caption color={Colors.Gray500}>No materializations</Caption>;
+  }
+
+  const {metadataEntries, timestamp} = event;
+
+  return (
+    <DetailsTable>
+      <tbody>
+        {metadataEntries.map((entry) => (
+          <tr key={`metadata-${entry.label}`}>
+            <td>
+              <Mono>{entry.label}</Mono>
+            </td>
+            <td>
+              <Mono>
+                <MetadataEntry entry={entry} expandSmallValues={true} />
+              </Mono>
+            </td>
+            <td style={{opacity: 0.7}}>{entry.description}</td>
+          </tr>
+        ))}
+
+        {(observations || []).map((obs) => (
+          <React.Fragment key={obs.timestamp}>
+            {obs.metadataEntries.map((entry) => (
+              <tr key={`metadata-${obs.timestamp}-${entry.label}`}>
+                <td>{entry.label}</td>
+                <td>
+                  <MetadataEntry entry={entry} expandSmallValues={true} />
+                </td>
+                <td style={{opacity: 0.7}}>
+                  <Box flex={{gap: 8, alignItems: 'center'}}>
```

Comment:
I think a flex-start align will be more correct here, since the text contents are multiline. The icon might need a little top spacing to make it line up nicely with the line-height of the first line of text, though.

Reply:
Fixed! I put a bit of top spacing on to align with the first line of text 👍